### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**
- [DEMO LINK](https://Zelinskyi-Serhii.github.io/layout_calendar/)
- [TEST REPORT LINK](https://Zelinskyi-Serhii.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

File: src/styles/blocks/calendar.scss [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+$week-days:
+  "mon" 0,
+  "tues" $box-size + $gap-between-days,
+  "wed" $box-size * 2 + (2 * $gap-between-days),
+  "thurs" $box-size * 3 + (3 * $gap-between-days),
+  "fri" $box-size * 4 + (4 * $gap-between-days),
+  "sat" $box-size * 5 + (5 * $gap-between-days),
+  "sun" $box-size * 6 + (6 * $gap-between-days);
```
Comment (kirill-vypirovskyi) [inline src/styles/blocks/calendar.scss]: You can add variables in braces and then multiply sum by 2
```suggestion
  "wed" ($box-size  +  $gap-between-days) * 2,
  "thurs" $box-size * 3 + (3 * $gap-between-days),
  "fri" $box-size * 4 + (4 * $gap-between-days),
  "sat" $box-size * 5 + (5 * $gap-between-days),
  "sun" $box-size * 6 + (6 * $gap-between-days);
```

Comment (kirill-vypirovskyi) [reply src/styles/blocks/calendar.scss]: I think it looks better

File: src/index.html [hunk 2]
```diff
@@ -7,6 +7,44 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+
+
+    <div class="calendar">
+      <div class="calendar__day calendar__day--1 calendar__day--tues"></div>
```
Comment (kirill-vypirovskyi) [inline src/index.html]: In my opinion it will be better to add modificator to calendar block, not to the first element.

**Review by kirill-vypirovskyi — CHANGES_REQUESTED** [review]
GJ, but you should have one more for-loop for regulate month length.
You also didn't do checklist p.12, sort your css properties

**Review by kirill-vypirovskyi — APPROVED** [review]

File: src/index.html [hunk 3]
```diff
@@ -7,6 +7,44 @@
...
+      <div class="calendar__day calendar__day--22"></div>
+      <div class="calendar__day calendar__day--23"></div>
+      <div class="calendar__day calendar__day--24"></div>
+      <div class="calendar__day calendar__day--25"></div>
+      <div class="calendar__day calendar__day--26"></div>
+      <div class="calendar__day calendar__day--27"></div>
+      <div class="calendar__day calendar__day--28"></div>
+      <div class="calendar__day calendar__day--29"></div>
+      <div class="calendar__day calendar__day--30"></div>
+      <div class="calendar__day calendar__day--31"></div>
```
Comment (illiarerun) [inline src/index.html]: remove these modifiers, better use pesudo-classes for selecting needed elements

File: src/styles/blocks/calendar.scss [hunk 4]
```diff
@@ -0,0 +1,55 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  position: relative;
+  top: 50%;
+  transform: translateY(-50%);
+  margin: auto;
+  width: 710px;
```
Comment (illiarerun) [inline src/styles/blocks/calendar.scss]: use formula to calculate width of container

File: src/styles/blocks/calendar.scss [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    "mon" 0,
+    "tues" $box-size + $gap-between-days,
+    "wed" 2 * ($box-size + $gap-between-days),
+    "thurs" 3 * ($box-size + $gap-between-days),
+    "fri" 4 * ($box-size + $gap-between-days),
+    "sat" 5 * ($box-size + $gap-between-days),
+    "sun" 6 * ($box-size + $gap-between-days);
+
+  @each $day, $margin in $week-days {
+    &--day-#{$day} &__day--1 {
```
Comment (illiarerun) [inline src/styles/blocks/calendar.scss]: use pseudoclass here
```suggestion
    &--day-#{$day}  {
```

File: src/styles/blocks/calendar.scss [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  &__day {
+    cursor: pointer;
+    @include square($box-size);
+    background-color: $day-color;
+    border: 1px solid black;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition: 0.5s;
```
Comment (illiarerun) [inline src/styles/blocks/calendar.scss]: [checklist](https://github.com/mate-academy/layout_calendar/blob/master/checklist.md) 8th item

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 7]
```diff
@@ -7,6 +7,44 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+
+
+    <div class="calendar calendar--day-fri calendar--day-month-30">
```
Comment (sergii-nosachenko) [inline src/index.html]: Read the task carefully - you need to rename your modifiers.

Also checklist #2

File: src/index.html [hunk 8]
```diff
@@ -7,6 +7,44 @@
...
+      <div class="calendar__day"></div>
+      <div class="calendar__day"></div>
+      <div class="calendar__day"></div>
+      <div class="calendar__day"></div>
+      <div class="calendar__day"></div>
+    </div>
+
+
+
+
```
Comment (sergii-nosachenko) [inline src/index.html]:
```suggestion
```

Cleanup your code

File: src/styles/blocks/calendar.scss [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  gap: $gap-between-days;
+
+  $week-days:
+    "mon" 0,
+    "tues" $box-size + $gap-between-days,
+    "wed" 2 * ($box-size + $gap-between-days),
+    "thurs" 3 * ($box-size + $gap-between-days),
+    "fri" 4 * ($box-size + $gap-between-days),
+    "sat" 5 * ($box-size + $gap-between-days),
+    "sun" 6 * ($box-size + $gap-between-days);
```
Comment (sergii-nosachenko) [inline src/styles/blocks/calendar.scss]: Store this in variables.

Comment (sergii-nosachenko) [reply src/styles/blocks/calendar.scss]: Read the task carefully on how to name days correctly.

File: src/styles/blocks/calendar.scss [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    align-items: center;
+    transition: background-color $transition-time,
+      color $transition-time,
+      transform $transition-time
+    ;
+
+    &:hover {
+      background-color: pink;
+      color: yellowgreen;
+      transform: translate(0, -20px);
```
Comment (sergii-nosachenko) [inline src/styles/blocks/calendar.scss]:
```suggestion
      transform: translateY(-20px);
```

File: src/styles/blocks/calendar.scss [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    @include square($box-size);
+    background-color: $day-color;
+    border: 1px solid black;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition: background-color $transition-time,
+      color $transition-time,
+      transform $transition-time
+    ;
```
Comment (sergii-nosachenko) [inline src/styles/blocks/calendar.scss]: Checklist #12

File: src/styles/main.scss [hunk 12]
```diff
@@ -1,3 +1,9 @@
+@import "utils/variables";
+@import "utils/mixins";
+@import "utils/normalize";
+@import "blocks/calendar.scss";
+
 body {
+  height: 100vh;
   margin: 0;
 }
```
Comment (sergii-nosachenko) [inline src/styles/main.scss]: Make body as a separate BEM block with class.

File: src/styles/utils/_variables.scss [hunk 13]
```diff
@@ -0,0 +1,6 @@
+$day-color: #eee;
+$month-length: 31;
+$box-size: 100px;
+$gap-between-days: 1px;
+$transition-time: 0.3s;
```
Comment (sergii-nosachenko) [inline src/styles/utils/_variables.scss]: Wrong duration.

File: src/styles/blocks/calendar.scss [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+
+    &:hover {
+      background-color: pink;
+      color: yellowgreen;
```
Comment (sergii-nosachenko) [inline src/styles/blocks/calendar.scss]: Add colors to variables.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/styles/blocks/calendar.scss [hunk 15]
```diff
@@ -1,52 +1,52 @@
...
     display: flex;
     justify-content: center;
     align-items: center;
+    @include square($box-size);
+    background-color: $day-color;
+    border: 1px solid black;
+    cursor: pointer;
+
     transition: background-color $transition-time,
       color $transition-time,
```
Comment (sergii-nosachenko) [inline src/styles/blocks/calendar.scss]: There is no color prop to animate.

File: src/styles/blocks/calendar.scss [hunk 16]
```diff
@@ -1,52 +1,52 @@
 .calendar {
-  display: flex;
-  flex-wrap: wrap;
   position: relative;
   top: 50%;
-  transform: translateY(-50%);
+  display: flex;
+  flex-wrap: wrap;
   margin: auto;
```
Comment (sergii-nosachenko) [inline src/styles/blocks/calendar.scss]: Don't set margins to block itself. You could add styles to the parent element to center this block.

File: src/styles/blocks/calendar.scss [hunk 17]
```diff
@@ -1,52 +1,52 @@
...
+    "sun" 6 * $width-day;
 
   @each $day, $margin in $week-days {
-    &--day-#{$day} &__day:first-child {
+    &--start-day--#{$day} &__day:first-child {
       margin-left: $margin;
     }
   }
 
   @for $num from 28 through 31 {
```
Comment (sergii-nosachenko) [inline src/styles/blocks/calendar.scss]:
```suggestion
  @for $num from 28 through $month-length {
```

Maybe, you can add meaningful var for 28 too.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/styles/blocks/calendar.scss [hunk 18]
```diff
@@ -0,0 +1,54 @@
+.calendar {
+  position: absolute;
+  top: 50%;
+  display: flex;
+  flex-wrap: wrap;
+  width: $width-calendar;
+  gap: $gap-between-days;
+  transform: translateY(-50%);
```
Comment (sergii-nosachenko) [inline src/styles/blocks/calendar.scss]:
```suggestion
  display: flex;
  flex-wrap: wrap;
  width: $width-calendar;
  gap: $gap-between-days;
```

Overthinked positioning. Just remove these props at all.

File: src/styles/main.scss [hunk 19]
```diff
@@ -1,3 +1,5 @@
-body {
-  margin: 0;
-}
+@import "utils/variables";
+@import "utils/mixins";
+@import "utils/normalize";
+@import "blocks/page.scss";
+@import "blocks/calendar.scss";
```
Comment (sergii-nosachenko) [inline src/styles/main.scss]: You can omit `.scss` in imports.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

**Review by sergii-nosachenko — APPROVED** [review]
Good work 👍

**Pull request closed without merging**